### PR TITLE
Handle emoji codepoint size for some not so great terminals.

### LINKED ISSUE #1568: Printing inconsistent width characters and spacing
### What version are you using?

21.4.1

### What did you do?

Screenshot 2024-08-28 at 3 38 32 PM

### What did you expect to see?

Consistent spacing around emojis.

### What did you see instead?

Inconsistent spacing around emojis.

cc @fnando @elizabethengelman

### PR DESCRIPTION
**Pull request by fnando**
### What

VScode and Apple Terminal doesn't work well with emojis that have 2 codepoints; in this case, we need to add an extra space, so it works properly.

Fix #1568 

#### VScode

CleanShot 2024-09-03 at 16 01 21

CleanShot 2024-09-03 at 16 03 14

#### Apple Terminal

CleanShot 2024-09-03 at 16 04 18

CleanShot 2024-09-03 at 16 04 57

#### iTerm

CleanShot 2024-09-03 at 16 06 44

CleanShot 2024-09-03 at 16 05 51

### Why

So text alignment is consistent.

### Known limitations

N/A

### CODE REVIEW

File: cmd/soroban-cli/src/print.rs [hunk 1]
```diff
@@ -7,6 +7,8 @@ use crate::{
     utils::{explorer_url_for_transaction, transaction_hash},
 };
 
+const TERMS: &[&str] = &["Apple_Terminal", "vscode"];
```
Comment (leighmcculloch) [inline cmd/soroban-cli/src/print.rs:10]: How about windows?

Comment (fnando) [reply cmd/soroban-cli/src/print.rs:10]: Didn't test it yet, still need to fix my computer :/

Comment (Shaptic) [reply cmd/soroban-cli/src/print.rs:10]: Does cmd/powershell even support emojis?

Comment (fnando) [reply cmd/soroban-cli/src/print.rs:10]: powershell does, and that's what we officially support.

Comment (fnando) [reply cmd/soroban-cli/src/print.rs:10]: CleanShot 2024-09-03 at 16 52 44

**Review by leighmcculloch — APPROVED** [review]

File: cmd/soroban-cli/src/print.rs (hunk 1 above)
Comment (fnando) [reply cmd/soroban-cli/src/print.rs:10]: @leighmcculloch @Shaptic just tested `main` on windows terminal and Microsoft got it right! 👏 

CleanShot 2024-09-04 at 09 51 02

**Pull request merged**